CRM/Civicase/Service/CaseCategoryMenu.php

```diff
@@ -306,4 +306,44 @@ private function addSeparatorToTheLastMenuOf($parentId) {
     ]);
   }
 
+  /**
+   * Activates Case Category Main menu.
+   *
+   * @param string $caseTypeCategoryName
+   *   Case Type category name.
+   */
+  public function activateItem($caseTypeCategoryName) {
+    $result = civicrm_api3('Navigation', 'get', ['name' => $caseTypeCategoryName]);
+
+    if ($result['count'] === 0) {
+      return;
+    }
+
+    $navigationItem = array_shift($result['values']);
+    civicrm_api3('Navigation', 'create', [
+      'id' => $navigationItem['id'],
+      'is_active' => TRUE,
+    ]);
+  }
+
+  /**
+   * Deactivates Case Category Main menu.
+   *
+   * @param string $caseTypeCategoryName
+   *   Case Type category name.
+   */
+  public function deactivateItem($caseTypeCategoryName) {
+    $result = civicrm_api3('Navigation', 'get', ['name' => $caseTypeCategoryName]);
+
+    if ($result['count'] === 0) {
+      return;
+    }
+
+    $navigationItem = array_shift($result['values']);
+    civicrm_api3('Navigation', 'create', [
+      'id' => $navigationItem['id'],
+      'is_active' => FALSE,
+    ]);
+  }
+
```

Comment:
Instead of duplicating a lot of code for the activate and deactivate function, why don't we do something like this: ?
```php
  private function setStatus($caseTypeCategoryName, $status) {
    $result = civicrm_api3('Navigation', 'get', ['name' => $caseTypeCategoryName]);

    if ($result['count'] === 0) {
      return;
    }

    $navigationItem = array_shift($result['values']);
    civicrm_api3('Navigation', 'create', [
      'id' => $navigationItem['id'],
      'is_active' => FALSE,
    ]);
  }
  
  public function deactivateItem($caseTypeCategoryName){
    $this->setStatus($caseTypeCategoryName, FALSE);
  }

  public function activateItem($caseTypeCategoryName){
    $this->setStatus($caseTypeCategoryName, TRUE);
  }

```